zeppelin-server/src/main/java/org/apache/zeppelin/rest/NotebookRestApi.java

```diff
@@ -750,20 +762,31 @@ public Response runParagraphSynchronously(@PathParam("noteId") String noteId,
 
     // handle params if presented
     handleParagraphParams(message, note, paragraph);
-
-    if (paragraph.getListener() == null) {
+    if ( paragraph.getListener() == null){
       note.initializeJobListenerForParagraph(paragraph);
     }
-
     paragraph.run();
-
     final InterpreterResult result = paragraph.getResult();
-
-    if (result.code() == InterpreterResult.Code.SUCCESS) {
-      return new JsonResponse<>(Status.OK, result).build();
-    } else {
+    if ( result.code() != InterpreterResult.Code.SUCCESS){
       return new JsonResponse<>(Status.INTERNAL_SERVER_ERROR, result).build();
     }
+    InterpreterResult resultFollowers;
+    if ( subsequent ) {
+      List<Paragraph> paragraphList = note.getParagraphs();
+      for (Paragraph p : paragraphList.subList(paragraphList.indexOf(paragraph) + 1,
+                                               paragraphList.size())) {
+        if (p.getListener() == null) {
+          note.initializeJobListenerForParagraph(p);
+        }
+        p.run();
```

Comment:
calling `run()` from Paragraph directly should be avoided. Paragraph should always run through scheduler provided by interpreter. i.e `note.run(p.getId())`. 

But currently, there's no method that runs paragraph synchronously in class Note. This PR doesn't need to address, but later it's better implement method such as `note.runSynchronous(p.getId())` which submit job into scheduler one by one, instead of directly run Paragraph.run().